sprint_1

EvgeniyGrokholskiy opened the pull request.

EvgeniyGrokholskiy committed:
feature_(APP): add APP_08.08.23.

EvgeniyGrokholskiy committed:
feature_(README.md): change chat version_08.08.23.

EvgeniyGrokholskiy committed:
fix_(types.d.ts): add newline at end_08.08.23.

EvgeniyGrokholskiy committed:
fix_(newline): add newline at end_08.08.23.

EvgeniyGrokholskiy committed:
fix_(newline): add newline at end_08.08.23.

EvgeniyGrokholskiy committed:
fix_(newline): add newline at end_08.08.23.

EvgeniyGrokholskiy committed:
fix_(chatPage.ts): remove unused code_08.08.23.

gennady-bars commented on src/index.html:
```diff
@@ -0,0 +1,22 @@
+    <title>Vite App</title>
+  </head>
+  <body>
+    <div class='page_wrapper link_block'>
```
Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту. Будет `nav`, потом `ul`, а потом уже `li`.

gennady-bars commented on src/index.html:
```diff
@@ -0,0 +1,22 @@
+  </head>
+  <body>
+    <div class='page_wrapper link_block'>
+      <a href='pages/login/login.html' class='font_20'>Login page</a>
+      <a href='pages/registration/registration.html' class='font_20'>Registration page</a>
+      <a href='pages/chat/chat.html' class='font_20'>Chat page</a>
+      <a href='pages/userProfile/userProfile.html' class='font_20'>User profile page</a>
+      <a href='pages/5XX/5XX.html' class='font_20'>5XX page</a>
+      <a href='pages/404/404.html' class='font_20'>404 page</a>
+      <a href='pages/uiLib/uiLib.html' class='font_20'>UI Lib page</a>
```
Список ссылок выглядит как список, значит, и отрисовываться должен с помощью `ul` и `li`. Это стандартная практика, чтобы улучшить `UX` (это нужно для парсеров сайтов и для спец возможностей браузера). A внутри `li` уже тег `a` будет.

gennady-bars commented on src/index.html:
```diff
@@ -0,0 +1,22 @@
+      <a href='pages/404/404.html' class='font_20'>404 page</a>
+      <a href='pages/uiLib/uiLib.html' class='font_20'>UI Lib page</a>
+    </div>
+    <div id='app'></div>
```
**Можно лучше**

Чтобы сделать сайт более доступным для пользователя, нужно стараться придерживаться **семантической верстки**.
Несколько полезных статей на эту тему:
•	[Что такое семантическая вёрстка и зачем она нужна](https://htmlacademy.ru/blog/html/semantics)
•	[Семантика для циников](https://www.youtube.com/watch?v=B5pomLr059U)

gennady-bars commented on src/variables.scss:
```diff
@@ -0,0 +1,14 @@
+$common-black-text: #1E1E1E;
```
* Отлично, что цвета вынесены в `css`-константы

gennady-bars commented on src/partials/layouts/page.hbs:
```diff
@@ -0,0 +1,14 @@
+    <title>{{title}}</title>
+</head>
+<body>
+    <main>
```
* Отлично, что не забываете про тег `main` для главной части страницы

gennady-bars commented on src/partials/chatItem/chatItem.hbs:
```diff
@@ -0,0 +1,26 @@
+    <div class='chat-item_inner-wrapper {{selected}}'>
+        <div class='chat-item-avatar-wrapper'>
+            {{#if avatarUrl}}
+                <img src='{{avatarUrl}}' alt='avatar' height='47' width='47' />
```
* Отлично, что не забываете про `alt` в тегах `img`

gennady-bars commented on server.js:
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+
+const app = express();
+const PORT = 3000;
```
**Можно лучше**

Лучше дать возможность поменять порт из консоли, если понадобится: `const PORT = process.env.PORT || 3000;`  . Если не задан порт, тогда будет `3000` по умолчанию.

EvgeniyGrokholskiy replied:
Сделал

gennady-bars commented on package.json:
```diff
@@ -0,0 +1,27 @@
+    "node": ">=12"
+  },
+  "scripts": {
+    "dev": "vite",
```
**Можно лучше**

Чтобы **автоматически запускался** проект в браузере, можно дописать в скрипт `--open`

gennady-bars reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на **гитхабе** все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная:

* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* Отлично, что цвета вынесены в `css`-константы
* Отлично, что не забываете про `alt` в тегах `img`
* Отлично, что не забываете про тег `main` для главной части страницы




но есть некоторые недочеты:

*  Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту. Будет `nav`, потом `ul`, а потом уже `li`.
* Список ссылок выглядит как список, значит, и отрисовываться должен с помощью `ul` и `li`. Это стандартная практика, чтобы улучшить `UX` (это нужно для парсеров сайтов и для спец возможностей браузера). A внутри `li` уже тег `a` будет.



Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте **работоспособность проекта** и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

Удачного рефакторинга кода.

EvgeniyGrokholskiy committed:
fix_(src/index.html): fix fixed errors in semantic layout_08.08.23

	package.json: add --open in dev script
	server.js: add process.env.PORT || 3000
	registrationPage.ts: fix error in console,
	two <input /> with same id on page

gennady-bars reviewed (approved):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
